### PR TITLE
Adds popular() method to Client

### PR DESCRIPTION
**Pull request by JordanAdams**
Allows for pulling the current most popular packages

**Usage:**
```
$client->popular($totalResults)
```

### CODE REVIEW

File: src/Packagist/Api/Result/Factory.php [hunk 1]
```diff
@@ -41,7 +43,7 @@ public function create(array $data)
      */
     public function createSearchResults(array $results)
     {
-        $created = array();
+        $created = [];
```
Comment (GrahamCampbell) [inline src/Packagist/Api/Result/Factory.php]: Don't change these

Comment (JordanAdams) [inline src/Packagist/Api/Result/Factory.php]: Automatically done by my PHPStorm. I'll fix it now :+1:

File: spec/Packagist/Api/Result/FactorySpec.php [hunk 2]
```diff
@@ -23,6 +23,18 @@ function it_creates_search_results()
         }
     }
 
+	function it_creates_popular_results()
```
Comment (umpirsky) [inline spec/Packagist/Api/Result/FactorySpec.php]: Please use 4 spaces for tab.

File: spec/Packagist/Api/ClientSpec.php [hunk 3]
```diff
@@ -45,6 +45,16 @@ function it_searches_for_packages_with_filters(HttpClient $client, Factory $fact
         $this->search('sylius', array('tag' => 'storage'));
     }
 
+	function it_gets_popular_packages(HttpClient $client, Factory $factory, Request $request, Response $response)
```
Comment (umpirsky) [inline spec/Packagist/Api/ClientSpec.php]: Please use 4 spaces for tab.

File: src/Packagist/Api/Result/Factory.php [hunk 4]
```diff
@@ -22,7 +22,9 @@ class Factory
     public function create(array $data)
     {
         if (isset($data['results'])) {
-            return $this->createSearchResults($data['results']);
+			return $this->createSearchResults($data['results']);
```
Comment (umpirsky) [inline src/Packagist/Api/Result/Factory.php]: Please use 4 spaces for tab.

**Comment by umpirsky** [comment]
@JordanAdams Can you please add `popular.php` to https://github.com/KnpLabs/packagist-api/tree/master/examples? Also, build is failing for 5.3.

Thanks for your contribution, great job so far! :+1:

**Comment by JordanAdams** [comment]
I'm considering tweaking this to accept the number of results required instead eg. `$client->popular(300)` would return the top 300 packages. Would this be preferred?

**Comment by umpirsky** [comment]
Not sure, what do you think @docteurklein ?

**Comment by JordanAdams** [comment]
Should be all good now

**Comment by umpirsky** [comment]
@JordanAdams Awesome, thanks! :+1:

**Pull request merged**